src/sql/src/statements.rs

```diff
@@ -133,6 +140,32 @@ fn parse_string_to_value(
     }
 }
 
+/// Casts string to value of specified numeric data type.
+/// If the string cannot be parsed, returns the default value for the numeric type.
+///
+/// Returns None if the data type doesn't support auto casting.
+fn auto_cast_to_numeric(s: &str, data_type: &ConcreteDataType) -> Option<Value> {
+    let value = match data_type {
+        ConcreteDataType::Boolean(_) => Value::Boolean(s.parse::<bool>().unwrap_or_default()),
```

Comment:
Why not return error (or do nothing, pass through) if the casting is failed? Using default value could cause surprise to users.

Reply:
> Why not return error (or do nothing, pass through) if the casting is failed? Using default value could cause surprise to users.

IMHO here we should return  `Value::Null` instead of the default value of target data type.

Reply:
Maybe MySQL changed this behavior. I can follow the latest behavior.